test/testOffline.js

```diff
@@ -257,6 +257,31 @@ describe('Offline', function() {
     });
   });
 
+  it('should cache directoryIndexes as dirs', function() {
+    var rootDir = temp.mkdirSync('oghliner');
+    var dir = path.join(rootDir, 'dist');
+    fs.mkdirSync(dir);
+    var subDir = path.join(dir, 'subdir');
+    fs.mkdirSync(subDir);
+
+    fs.writeFileSync(path.join(subDir, 'test_file_1.js'), 'test_file_1');
+    fs.writeFileSync(path.join(subDir, 'test_file_2.js'), 'test_file_2');
+    fs.writeFileSync(path.join(subDir, 'test_file_3.js'), 'test_file_3');
+
+    process.chdir(rootDir);
+
+    return offline({
+      rootDir: dir,
+      directoryIndexes: ['test_file_1.js']
+    }).then(function() {
+      var content = fs.readFileSync(path.join(dir, 'offline-worker.js'), 'utf8');
+      assert.notEqual(content.indexOf('/'), -1, 'Must have a directory in the output');
```

Comment:
This won't actually ensure that there's a directory in the output, since offline-worker.js is full of forward slashes anyway. Looking for `./` would be more reliable.

But that reminds me that offline-worker.js already caches `./` unconditionally. With *directoryIndexes* set to `['index.html']` and a top-level index.html file, that creates a duplicate entry for `./` in the worker's *RESOURCES* array.

It seems like we should drop that unconditional entry and instead set the default value of *directoryIndexes* to `['index.html']`, which seems by far the most likely value (given that it's the default *DirectoryIndex* for the servers that host GitHub Pages.)